test/packages/good/LICENSE.txt

```diff
@@ -0,0 +1,202 @@
+      except as required for reasonable and customary use in describing the
+      origin of the Work and reproducing the content of the NOTICE file.
+
+   7. Disclaimer of Warranty. Unless required by applicable law or
+      agreed to in writing, Licensor provides the Work (and each
+      Contributor provides its Contributions) on an "AS IS" BASIS,
+      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+      implied, including, without limitation, any warranties or conditions
+      of TITLE, NON-INFRINGEMENT, MERCHANTABILITY, or FITNESS FOR A
+      PARTICULAR PURPOSE. You are solely responsible for determining the
+      appropriateness of using or redistributing the Work and assume any
+      risks associated with Your exercise of permissions under this License.
+
+   8. Limitation of Liability. In no event and under no legal theory,
+      whether in tort (including negligence), contract, or otherwise,
+      unless required by applicable law (such as deliberate and grossly
+      negligent acts) or agreed to in writing, shall any Contributor be
+      liable to You for damages, including any direct, indirect, special,
+      incidental, or consequential damages of any character arising as a
+      result of this License or out of the use or inability to use the
+      Work (including but not limited to damages for loss of goodwill,
+      work stoppage, computer failure or malfunction, or any and all
+      other commercial damages or losses), even if such Contributor
+      has been advised of the possibility of such damages.
+
+   9. Accepting Warranty or Additional Liability. While redistributing
+      the Work or Derivative Works thereof, You may choose to offer,
+      and charge a fee for, acceptance of support, warranty, indemnity,
+      or other liability obligations and/or rights consistent with this
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
+
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```

Comment:
It looks to me more like a part of the building procedure:
- you could use data from the manifest to populate the license
- it's more advanced than just formatting.
- user can define their own custom license, store it in `_dev/build` like docs templates.

Reply:
But take into account that it is not only about populating these fields, in principle the Apache License requires headers in licensed files, this may be difficult to support here.

Also we don't have enough information in packages now, at least we are missing the copyright owner now, we only have a github handle, for example for the "Apache" package the github handle is now `elastic/obs-service-integrations`, but the copyright owner should be "Elasticsearch B.V.".

Another option I see is:
* The `source.license` field is only an indication for package consumers, it is responsibility of package developers to properly apply the licenses.
* Developers may use tools like `go-licenser` to properly apply the licenses to their work, or do it manually, as they prefer.
  * `elastic-package format` could be this tool if `source.license` is set and we add a field in the manifest for the copyright owner. License text could be still included in build time.

> user can define their own custom license, store it in `_dev/build` like docs templates.

In principle we were considering allowing only a limited set of licenses, see the mentions to an enum [here](https://github.com/elastic/package-spec/issues/298#issue-1172038299) and preference over arbitrary text files [here](https://github.com/elastic/package-spec/issues/298#issuecomment-1076305534).

Or do you mean to include here the template for the headers of the files?

Reply:
> Also we don't have enough information in packages now, at least we are missing the copyright owner now, we only have a github handle, for example for the "Apache" package the github handle is now elastic/obs-service-integrations, but the copyright owner should be "Elasticsearch B.V.".

For such cases, we can assume default field values (for example "Elasticsearch B.V.") to populate the license template.

> Or do you mean to include here the template for the headers of the files?

Could you please provide a sample header that needs customization? If we want to support a limited set of licenses, those can be for sure embedded in elastic-package. I'm still wondering if this is something we can render out of embedded templates. Even if we have to add more properties to the manifest.

Reply:
> > Also we don't have enough information in packages now, at least we are missing the copyright owner now, we only have a github handle, for example for the "Apache" package the github handle is now elastic/obs-service-integrations, but the copyright owner should be "Elasticsearch B.V.".
> 
> For such cases, we can assume default field values (for example "Elasticsearch B.V.") to populate the license template.

How would we support external developers (community, partners...). Or we delay this by now?

> > Or do you mean to include here the template for the headers of the files?
> 
> Could you please provide a sample header that needs customization? If we want to support a limited set of licenses, those can be for sure embedded in elastic-package. I'm still wondering if this is something we can render out of embedded templates. Even if we have to add more properties to the manifest.

With customizations I mean here for example filling the placeholders in the header for the Apache license.

Reply:
> How would we support external developers (community, partners...). Or we delay this by now?

Yes, as we want to start with a predefined set of licenses, I meant postponing the community area.